tsdb/meta.go

```diff
@@ -16,13 +15,203 @@ const (
 	maxStringLength = 64 * 1024
 )
 
+// Database is the in memory index of a collection of measurements, time series, and their tags.
+type Database struct {
+	// in memory metadata index, built on load and updated when new series come in
```

Comment:
Should it be called `MetaIndex`?  The comment above and below the `type` describe it that way.